frontend/packages/client/src/components/Community/CommunityEditorLinks.js

```diff
@@ -1,117 +1,21 @@
 import { WrapperResponsive, Loader } from "components";
+import useLinkValidator from "./hooks/useLinkValidator";
 
-export default function CommunityEditorLinks(props = {}) {
-  const {
-    websiteUrl = "",
-    twitterUrl = "",
-    instagramUrl = "",
-    discordUrl = "",
-    githubUrl = "",
-    updateCommunity,
-  } = props;
-  const [enableSave, setEnableSave] = useState(false);
-  const [isUpdating, setIsUpdating] = useState(false);
-  const [links, setLinks] = useState({
-    websiteUrl,
-    twitterUrl,
-    instagramUrl,
-    discordUrl,
-    githubUrl,
-  });
-
-  const saveData = async () => {
-    setIsUpdating(true);
-    const updatedKeys = Object.keys(links).filter(
-      (key) => links[key] !== (props[key] ?? "")
-    );
-
-    const updatedFields = Object.assign(
-      {},
-      ...updatedKeys.map((key) => ({ [key]: links[key] }))
-    );
-    await updateCommunity(updatedFields);
-    setIsUpdating(false);
-  };
-
-  const urlPatternValidation = (url) => {
-    const regex = new RegExp(
-      "^(https?:\\/\\/)?" + // protocol
-        "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // domain name
-        "((\\d{1,3}\\.){3}\\d{1,3}))" + // OR ip (v4) address
-        "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
-        "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
-        "(\\#[-a-z\\d_]*)?$",
-      "i"
-    );
-    return url === "" || !!regex.test(url);
-  };
-  const twitterValidator = (url) => {
-    const regex = new RegExp(
-      "^(?:https://)?(?:www\\.)?twitter\\.com/(\\w+)",
-      "i"
-    );
-    return url === "" || !!regex.test(url);
-  };
-  const instagramValidator = (url) => {
-    return (
-      url === "" ||
-      /(?:(?:http|https):\/\/)?(?:www.)?(?:instagram.com|instagr.am|instagr.com)\/(\w+)/gim.test(
-        url
-      )
-    );
-  };
-  const discordValidator = (url) => {
-    return (
-      url === "" ||
-      /(https?:\/\/)?(www\.)?(discord\.(gg|io|me|li|com)|discordapp\.com\/invite)\/.+[a-z]/gim.test(
-        url
-      )
-    );
-  };
-  const githubValidator = (url) => {
-    return url === "" || /https?:\/\/github\.com\/(?:[^/\s]+)/gim.test(url);
-  };
-
-  const urlLinkNotRepeated = useCallback(() => {
-    const values = Object.values(links).filter((link) => link !== "");
-
-    const valuesSet = new Set(values);
-
-    if (valuesSet.size === values.length) {
-      return true;
-    }
-    return false;
-  }, [links]);
-
-  useEffect(() => {
-    if (
-      Object.keys(links).some((key) => links[key] !== (props[key] ?? "")) &&
-      urlPatternValidation(links["websiteUrl"]) &&
-      twitterValidator(links["twitterUrl"]) &&
-      instagramValidator(links["instagramUrl"]) &&
-      discordValidator(links["discordUrl"]) &&
-      githubValidator(links["githubUrl"]) &&
-      urlLinkNotRepeated()
-    ) {
-      setEnableSave(true);
-    } else {
-      setEnableSave(false);
-    }
-  }, [links, props, urlLinkNotRepeated]);
-
+export const CommunityLinksForm = ({
```

Comment:
converted this one into a reusable component used in edit community and create community